File: python/7 kyu/118.py
# https://www.codewars.com/kata/5470ae03304c1250b4000e57/train/python

# Passed

def identify_weapon(character):
    character_weapon_dict = {
    character_weapon.split("-")[0].strip(): character_weapon.strip() for character_weapon in 
     '''Laval-Shado Valious
        Cragger-Vengdualize
        Lagravis-Blazeprowlor
        Crominus-Grandorius
        Tormak-Tygafyre
        LiElla-Roarburn'''.splitlines()
    }
    weapon = character_weapon_dict.get(character, "Not a character")
    return weapon

output = identify_weapon("Crominus")
print(output)

File: nodejs/7 kyu/118.js
// https://www.codewars.com/kata/55563df50dda59adf900004d/train/javascript

// Passed

function validateEAN(eanCode) {
  let checksum = Number(eanCode[eanCode.length - 1]);
  let sumOfOperations = eanCode
    .slice(0, eanCode.length - 1)
    .split("")
    .map((stringDigit, idx) => Number(stringDigit) * (idx % 2 === 0 ? 1 : 3))
    .reduce((prev, curr) => prev + curr);
  let calculatedChecksum =
    sumOfOperations % 10 === 0 ? 0 : 10 - (sumOfOperations % 10);
  return checksum === calculatedChecksum;
}

const output = validateEAN("9783815820865");
console.log(output);